Dockerfile-base

```diff
@@ -1,41 +1,63 @@
+
 
+# Install qpdf
+# ------------
 WORKDIR /opt
 RUN wget https://s3.amazonaws.com/sharelatex-random-files/qpdf-6.0.0.tar.gz && tar xzf qpdf-6.0.0.tar.gz
 WORKDIR /opt/qpdf-6.0.0
 RUN ./configure && make && make install && ldconfig
 
+
+# Install Node
+# ------------
+RUN curl -sL https://deb.nodesource.com/setup_10.x | sudo bash -
+RUN apt-get install -y nodejs
+RUN npm install -g grunt-cli
+
+
+# Install Node6 (required by some services)
+# -----------------------------------------
+RUN wget https://nodejs.org/dist/v6.17.1/node-v6.17.1-linux-x64.tar.gz && \
+	mkdir -p /opt/nodejs && tar -xzf node-v6.17.1-linux-x64.tar.gz -C /opt/nodejs/ && \
+    cd /opt/nodejs && mv node-v6.17.1-linux-x64 6.17.1 && \
+    ln -s /opt/nodejs/6.17.1/bin/node /usr/bin/node6
+
+
 # Install TexLive
-RUN wget http://mirror.ctan.org/systems/texlive/tlnet/install-tl-unx.tar.gz; \
+# ---------------
+RUN wget https://ctan.crest.fr/tex-archive/systems/texlive/tlnet/install-tl-unx.tar.gz; \
 	mkdir /install-tl-unx; \
 	tar -xvf install-tl-unx.tar.gz -C /install-tl-unx --strip-components=1
-
 RUN echo "selected_scheme scheme-basic" >> /install-tl-unx/texlive.profile; \
-	/install-tl-unx/install-tl -profile /install-tl-unx/texlive.profile
-
+	/install-tl-unx/install-tl -profile /install-tl-unx/texlive.profile \
+	-repository http://tug.ctan.org/systems/texlive/tlnet/
 RUN rm -r /install-tl-unx; \
 	rm install-tl-unx.tar.gz
-
-ENV PATH /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/texlive/2017/bin/x86_64-linux/
+ENV PATH /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/texlive/2019/bin/x86_64-linux/
 RUN tlmgr install latexmk
 RUN tlmgr install texcount
 
-RUN npm install -g grunt-cli
 
 # Set up sharelatex user and home directory
+# -----------------------------------------
 RUN adduser --system --group --home /var/www/sharelatex --no-create-home sharelatex; \
 	mkdir -p /var/lib/sharelatex; \
 	chown www-data:www-data /var/lib/sharelatex; \
 	mkdir -p /var/log/sharelatex; \
 	chown www-data:www-data /var/log/sharelatex; \
 	mkdir -p /var/lib/sharelatex/data/template_files; \
-	chown www-data:www-data /var/lib/sharelatex/data/template_files;
-
+	chown www-data:www-data /var/lib/sharelatex/data/template_files;
```

Comment:
Nitpick: could you set up your editor to leave the trailing newlines in? (Or better yet copy the `.editorconfig` from v1 to this repo and use that, so it will be consistent for everyone.)